src/tools/pbs_upgrade_job.c

```diff
@@ -194,73 +197,57 @@ check_job_file(int fd)
 {
 	off_t pos_saved;
 	off_t pos_new;
-	int ret = -1;
-	int len;
+	int ret_version = -1;
+	int length = -1;
+	jobfix_PRE19 old_jobfix_pre19;
 
 	/* Save our current position so we can comeback to it */
 	pos_saved = lseek(fd, 0, SEEK_CUR);
 
-	/* The string "Job_Name" should appear at address 01160 (octal) for v12 */
-	pos_new = lseek(fd, 01160, SEEK_SET);
-	if (pos_new != 01160) {
-		fprintf(stderr, "File format not recognized.\n");
-		goto check_job_file_exit;
+	/*---------- For PBSPro >=13.x or <=18.x versions jobfix structure ---------- */
+	pos_new = lseek(fd, pos_saved, SEEK_SET);
+	if (pos_new != 0) {
+			fprintf(stderr, "Couldn't set the file position to zero.\n");
+			goto check_job_file_exit;
 	}
-	errno = 0;
-	len = read(fd, &buf, 8);
-	if (len < 0) {
+	memset(&old_jobfix_pre19.ji_jsversion, 0, sizeof(old_jobfix_pre19.ji_jsversion));
```

Comment:
Instead of memset just initialise ji_jsversion to zero.

Reply:
done!.